pkg/operator/controllers/condition.go

```diff
@@ -25,7 +25,8 @@ func SetCondition(ctx context.Context, arocli aroclient.Interface, cond *status.
 
 		changed := cluster.Status.Conditions.SetCondition(*cond)
 
-		if setStaticStatus(cluster, role) {
```

Comment:
before we remove this I would like really understand why it was there in the first place and what is the impact of removal?

Reply:
I went through the code and I am pretty sure it is safe to remove. The operator framework does a good job with updating of the status. I will do one more try tomorrow, just to triple check.